src/XR/features/WebXRDOMOverlay.ts

```diff
@@ -0,0 +1,139 @@
+import { Tools } from "../../Misc/tools";
+import { Nullable } from "../../types";
+import { WebXRFeatureName, WebXRFeaturesManager } from "../webXRFeaturesManager";
+import { WebXRSessionManager } from "../webXRSessionManager";
+import { WebXRAbstractFeature } from "./WebXRAbstractFeature";
+
+/**
+ * Options for DOM Overlay feature
+ */
+export interface IWebXRDOMOverlayOptions {
+    /**
+     * Fail the Web XR Session if DOM overlay isn't available (TODO: would like to mark optional as a feature parameter)
```

Comment:
This can be done using the features manager, by defining the feature as required (which is the default, actually)